addons/mail/static/src/discuss/core/common/thread_actions.js

```diff
@@ -10,6 +11,18 @@ import { _t } from "@web/core/l10n/translation";
 import { usePopover } from "@web/core/popover/popover_hook";
 
 threadActionsRegistry
+    .add("attachments", {
+        condition: (component) =>
+            component.thread?.model === "discuss.channel" &&
+            (!component.props.chatWindow || component.props.chatWindow.isOpen),
+        component: AttachmentPanel,
+        icon: "fa fa-fw fa-file",
+        iconLarge: "fa fa-fw fa-lg fa-file",
+        name: _t("Show Attachment List"),
+        nameActive: _t("Hide Attachment List"),
+        sequence: 25,
+        toggle: true,
+    })
```

Comment:
Not something to change in this PR, just pointing out an issue with some thread actions like this one:
The reliance on component props and getters is not good, because these thread actions have prerequisite on the component that can easily omitted, thus making them not working.
Also they might conflict with existing props that mean something (e.g. getter `thread` doesn't necessarily mean the thread being viewed, e.g. `Composer@thread` is the origin thread).

I'm still thinking on a better solution. We could use `setup()` in the meantime to document what it's expected for the component to have, and what it means like the getter `thread` here.